test/sql/alter/alter_col/test_not_null_multi_tran.test

```diff
@@ -0,0 +1,100 @@
+# name: test/sql/alter/not_null/test_not_null_multi_tran.test
+# description: Test Set/Drop NOT NULL in multiple transactions
+# group: [alter_col]
+
+statement ok
+PRAGMA enable_verification
+
+# Scenario #1, tran1 set not null, tran2 not able to insert
+statement ok con1
+CREATE TABLE t(i INTEGER, j INTEGER)
+
+statement ok con1
+BEGIN TRANSACTION
+
+statement ok con1
+ALTER TABLE t ALTER COLUMN j SET NOT NULL
+
+statement ok con2
+BEGIN TRANSACTION
+
+statement error con2
+INSERT INTO t VALUES(7777, NULL)
+
+statement ok con1
+INSERT INTO t VALUES(7, 7)
+
+statement ok con2
+ROLLBACK
+
+statement ok con1
+COMMIT
+
+query I con1
+SELECT i FROM t
+----
+7
+
+# Scenario #2, tran1 set not null, tran2 failed to commit
+statement ok con1
+DROP TABLE IF EXISTS t
+
+statement ok con1
+CREATE TABLE t(i INTEGER, j INTEGER)
+
+statement ok con1
+BEGIN TRANSACTION
+
+statement ok con2
+BEGIN TRANSACTION
+
+statement ok con2
+INSERT INTO t VALUES(7777, NULL)
+
+statement ok con1
+ALTER TABLE t ALTER COLUMN j SET NOT NULL
+
+statement ok con1
+COMMIT
+
+statement error con2
+COMMIT
+
+statement error con1
+INSERT INTO t VALUES(7777, NULL)
+
+query I con1
+SELECT COUNT(*) FROM  t
+----
+0
+
+# Scenario #3, cannot insert if one connection alter column
+statement ok con1
+DROP TABLE IF EXISTS t
+
+statement ok con1
+CREATE TABLE t(i INTEGER, j INTEGER)
+
+statement ok con1
+BEGIN TRANSACTION
+
+statement ok con1
+ALTER TABLE t ALTER COLUMN j SET NOT NULL
+
+statement ok con2
+BEGIN TRANSACTION
+
+statement error con2
+INSERT INTO t VALUES(1, 1)
+
+statement ok con1
+COMMIT
+
+statement ok con2
+COMMIT
+
+query I con1
+SELECT count(*) from t
```

Comment:
Can we add a test where we `DROP CONSTRAINT`, followed by inserting `NULL` values, followed by rolling back the dropping of the constraint?

Reply:
Oh, I have such test in [test/sql/alter/alter_col/test_not_null_in_tran.test](https://github.com/duckdb/duckdb/pull/4185/files#diff-e442cf794434022088f348adf059fd68482e532b77a96003f867ceafc1488338R153-R168)
Is that OK? As for multi-trans an alter-table will just block the other transaction.

Reply:
Yes that looks good, thanks.